src/main/java/com/urswolfer/gerrit/client/rest/http/GerritRestClient.java

```diff
@@ -274,11 +278,13 @@ private void setRequestBody(String requestBody, HttpRequestBase method, ContentT
             cookieStore.clear();
             return Optional.absent();
         }
-        if (authData.isHttpPassword()) {
+        if (authData.isHttpPassword() || githubOAuthDetected) {
```

Comment:
I'm wondering if we would like to move this flag into `LoginCache`, just as `getHostSupportsGerritAuth`? Have you thought about this? I've not looked into details yet.

Reply:
Thanks for the suggestion @uwolfer, let me look into that today.
I was also thinking about making the path detection more generic, as Gerrit supports many OAuth providers, not just GitHub. But I'll address that concern in a separate PR: I like to use GitHub PR pretty much like Gerrit changes: 1 PR = 1 thing = 1 commit.

Reply:
Done, it is actually better, so that I can assert the OAuth detection in tests as well. Thanks for the suggestion.